### ISSUE TITLE
HiveError: This should not happen.

### ISSUE THREAD

**Issue by alidaoud** [hivedb/hive#872]
**Steps to Reproduce**
I got this problem suddenly when i was developing an app without any changes related to the hive package, even on another device with old version of the same app.

**Stack Trace**
![Screen Shot 2022-01-05 at 18 10 02](https://user-images.githubusercontent.com/38518048/148244989-ebae72d8-07e2-4230-8e8f-edea88f98d48.png)

```
D/EGL_emulation(17159): eglMakeCurrent: 0x6eb2668870: ver 3 0 (tinfo 0x6e426b7b18) (first time)


E/flutter (17159): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: HiveError: This should not happen. Please open an issue on GitHub.


E/flutter (17159): #0      BinaryReaderImpl.readFrame
package:hive/…/binary/binary_reader_impl.dart:250

E/flutter (17159): #1      FrameHelper.framesFromBytes
package:hive/…/binary/frame_helper.dart:21

E/flutter (17159): #2      FrameIoHelper.framesFromFile
package:hive/…/io/frame_io_helper.dart:42

E/flutter (17159): <asynchronous suspension>


E/flutter (17159): #3      StorageBackendVm.initialize
package:hive/…/vm/storage_backend_vm.dart:86

E/flutter (17159): <asynchronous suspension>


E/flutter (17159): #4      HiveImpl._openBox
package:hive/src/hive_impl.dart:116

E/flutter (17159): <asynchronous suspension>


E/flutter (17159): #5      HiveImpl.openBox
package:hive/src/hive_impl.dart:145

E/flutter (17159): <asynchronous suspension>


E/flutter (17159): #6      SplashScreen.prepareAppUtils
package:store/…/splash_screen/splash.dart:122

E/flutter (17159): <asynchronous suspension>


E/flutter (17159):


E/flutter (17159): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: HiveError: This should not happen. Please open an issue on GitHub.


E/flutter (17159): #0      BinaryReaderImpl.readFrame
package:hive/…/binary/binary_reader_impl.dart:250

E/flutter (17159): #1      FrameHelper.framesFromBytes
package:hive/…/binary/frame_helper.dart:21

```

**Flutter Doctor**

```
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-arm, locale
    en-TR)
    • Flutter version 2.5.3 at /Users/alidaoud/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (3 months ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version
    32.0.0-rc1)
    • Android SDK at /Users/alidaoud/Library/Android/sdk
    • Platform android-31, build-tools 32.0.0-rc1
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.2.1, Build version 13C100
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (3 available)
    • Android SDK built for arm64 (mobile) • emulator-5554
      • android-arm64  • Android 11 (API 30) (emulator)
    • iPhone 13 (mobile)                   •
      D31B2746-78BD-4FDD-8E33-20A29DA224B5 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)                         • chrome
      • web-javascript • Google Chrome 96.0.4664.110

• No issues found!

```

**Version**
 - Platform: Android (still working on iOS though)
 - Flutter version: 2.5.3
 - Hive version: 2.0.4

**Comment by zmoshansky** [comment]
I had the same issue. See details here: https://github.com/felangel/bloc/issues/3243.

**Comment by themisir** [comment]
Closed with #914.

Please upgrade to hive [v2.0.6](https://pub.dev/packages/hive/versions/2.0.6).
